src/client/java/aztech/modern_industrialization/machines/gui/MachineScreen.java

```diff
@@ -211,10 +211,15 @@ protected void renderBg(GuiGraphics guiGraphics, float delta, int mouseX, int mo
 
     private void renderConfigurableSlotBackgrounds(GuiGraphics guiGraphics) {
         for (Slot slot : this.menu.slots) {
-            if (slot instanceof BackgroundRenderedSlot brs) {
+            if (slot.isActive() && slot instanceof BackgroundRenderedSlot brs) {
                 int px = leftPos + slot.x - 1;
                 int py = topPos + slot.y - 1;
-                guiGraphics.blit(SLOT_ATLAS, px, py, brs.getBackgroundU(), brs.getBackgroundV(), 18, 18);
+                if (slot.getItem().isEmpty()) {
```

Comment:
The added isEmpty check is so that the standard slot texture is rendered when an item is in the slot, but if there isn't an item, the designated slot texture is rendered instead. This makes it so that if the slot texture doesn't fully align with all items it may accept, it wont still render it behind the item if there is an item there.

As for the isActive check, I can't find my use case for it... it is a logical check to have there regardless I think. I may have had a need for it at one point and then changed something, not sure.

Reply:
I don't think the default values can be accessed using `super` here?

Reply:
Checking for isActive is a bad idea I think, maybe some background is still needed? Could make it so that returning a null atlas skips rendering if you want.

Reply:
And same for isEmpty, I'd rather give full control to the slot. For example fluid slots are always empty I think.